qemu_x86 and qemu_cortex_m3 time handling broken with CONFIG_QEMU_ICOUNT

Something is clearly wrong with qemu_x86 and qemu_cortex_m3 time handling. This issue is **not** present when using qemu_x86_64.

Setting `CONFIG_QEMU_ICOUNT=n` fixes the issue.

I've added the description below for the issue I saw, but he simplest really is to build `samples/philosophers` for qemu_x86 and see how real-time is not respected.

```
        k_sem_init(&sync_sem, 0, 1);
[...]
        printk("uptime 1: %lld\n", k_uptime_get());
        err = k_sem_take(&sync_sem, K_SEC(10));
        printk("uptime 2: %lld\n", k_uptime_get());
        BT_ASSERT_MSG(err == 0, "k_sem_take failed with err %d", err);
```

The code executes immediately, and the result is:

```
$ time west build -t run
-- west build: running target run
[3/10] Linking C executable zephyr/zephyr_prebuilt.elf
Memory region         Used Size  Region Size  %age Used
             RAM:      119516 B         4 MB      2.85%
        IDT_LIST:         464 B         2 KB     22.66%
[9/10] To exit from QEMU enter: 'CTRL+a, x'[QEMU] CPU: qemu32,+nx,+pae
SeaBIOS (version rel-1.12.1-0-ga5cab58-dirty-20200516_152813-e7ee184eb100-zephyr
)
Booting from ROM..Optimal CONFIG_X86_MMU_PAGE_POOL_PAGES 7
*** Booting Zephyr OS build zephyr-v2.3.0-415-g3f1991804144  ***
Starting Beacon Demo
uptime 1: 0
uptime 2: 10010
ASSERTION FAIL [err == 0] @ WEST_TOPDIR/zephyr/subsys/bluetooth/host/hci_core.c:372
        k_sem_take failed with err -11
[00:00:10.010,000] <err> os: EAX: 0x00109d59, EBX: 0x00109d59, ECX: 0x0000000a, EDX: 0x0011c72c
[00:00:10.010,000] <err> os: ESI: 0xfffffff5, EDI: 0x0000000a, EBP: 0x0011c764, ESP: 0x0011c72c
[00:00:10.010,000] <err> os: EFLAGS: 0x00000206 CS: 0x0008 CR3: 0x0011a180
[00:00:10.010,000] <err> os: call trace:
[00:00:10.010,000] <err> os: EIP: 0x00105092
[00:00:10.010,000] <err> os:      0x00105b94 (0xc03)
[00:00:10.010,000] <err> os:      0x00105fe4 (0x1)
[00:00:10.010,000] <err> os:      0x00100ef9 (0x11d108)
[00:00:10.010,000] <err> os:      0x00100eb4 (0x11a4e0)
[00:00:10.010,000] <err> os: >>> ZEPHYR FATAL ERROR 3: Kernel oops on CPU 0
[00:00:10.010,000] <err> os: Current thread: 0x0011a500 (unknown)
[00:00:10.010,000] <err> os: Halting system
FAILED: zephyr/CMakeFiles/run
cd /home/carles/src/zephyr/zephyr/build && /home/carles/bin/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/qemu-system-i386 -m 9 -cpu qemu32,+nx,+pae -device isa-debug-exit,iobase=0xf4,iosize=0x04 -no-reboot -nographic -net none -pidfile qemu.pid -chardev stdio,id=con,mux=on -serial chardev:con -mon chardev=con,mode=readline -serial unix:/tmp/bt-server-bredr -icount shift=5,align=off,sleep=off -rtc clock=vm -kernel /home/carles/src/zephyr/zephyr/build/zephyr/zephyr.elf
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /home/carles/bin/cmake/cmake-3.13.1-Linux-x86_64/bin/cmake --build /home/carles/src/zephyr/zephyr/build --target run

real    0m0.641s
user    0m0.502s
sys     0m0.163s

```

--- a/boards/Kconfig
+++ b/boards/Kconfig
@@ -31,7 +31,7 @@ menu "Board Options"
 config QEMU_ICOUNT
 	bool "QEMU icount mode"
 	depends on QEMU_TARGET
-	default y if !NETWORKING
+	default y if !NETWORKING && !BT
 	help
 	  Enable QEMU virtual instruction counter. The virtual cpu will
 	  execute one instruction every 2^N ns of virtual time. This will

